src/variant/variant.cpp

```diff
@@ -276,6 +284,24 @@ Variant::operator uint32_t() const {
 	return static_cast<uint32_t>(operator int64_t());
 }
 
+Variant::operator uint16_t() const {
+	return static_cast<uint16_t>(operator int64_t());
+}
+
+Variant::operator uint8_t() const {
+	return static_cast<uint8_t>(operator int64_t());
+}
+
+#ifdef NEED_LONG_INT
+Variant::operator signed long() const {
+	return static_cast<signed long>(operator int64_t());
+}
+
+Variant::operator unsigned long() const {
+	return static_cast<unsigned long>(operator int64_t());
+}
+#endif
```

Comment:
Since operators for both 64 and 32-bit ints are already defined this part is unnecessary and will cause conflicts (the same part makes sense in the engine, since there's no `uint32_t`/`int32_t` operators, and are operators for short/int).